### DISCUSSION TITLE
Cache and env variables

### DISCUSSION THREAD

**Discussion by ViliamKopecky** [vercel/turborepo#211] [Ideas]
I know I can provide list of .env files, but what about env variables provided from the machine. Local env vars and Vercel secrets.

- The build output will vary based on these values, so it should be included in the hash. Is that somehow possible?
- Leaking the artifacts as-is with secrets is a security issue. Did you think about encrypting it?
  - generate hash from sources
  - encrypt artifacts with that hash
  - generate second hash from encrypted artifacts and use it for storage and lookup.

**Comment by jaredpalmer** [comment]
Great question. Working on some new configuration keys that will help with this issue here: https://github.com/vercel/turborepo/blob/b95a104e727fda4394390ff6b4b93535e89b6fad/cli/internal/fs/turbo_json.go#L9-L16

Hoping to land this week as part of #218 

Similar to what you've suggest, we are exploring "Bring Your Own Key Encryption" with our security team as well. We think this is an awesome feature worth investing in.

**Comment by NixBiks** [comment]
@jaredpalmer  What's the recommended way to have multiple environments currently? E.g. local, staging and production.

I'm concerned about building locally and then staging and production will use environment variables used from that build stage cache. E.g. NextJS apps needs environment variables at build time...
